lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py

```diff
@@ -906,35 +903,43 @@ def exec_command_output(self, command):
         Returns:
             output of command execution
         """
-        data = self._data
-        data["command"] = '"' + command + '"'
-        param_list = shlex.split(self._command.format(**data))
+        param_list = shlex.split(self._command.format(**self._credentials, command=f'"{command}"'))
         return subprocess.check_output(
             param_list,
             shell=False,
         ).decode("utf-8")
 
+    def exec_command_capture_output(self, command):
+        """Run a command remotely on magma_dev VM.
+
+        Args:
+            command: command (str) to be executed on remote host
+            e.g. 'sed -i \'s/config1/config2/g\' /etc/magma/mme.yml'
+
+        Returns:
+            output of command execution
+        """
+        param_list = shlex.split(self._command.format(**self._credentials, command=f'"{command}"'))
+        return subprocess.run(
+            param_list,
+            shell=False,
+            capture_output=True,
+        )
+
     def detect_init_system(self) -> InitMode:
         """Detect whether services are running with Docker or systemd."""
-        def _is_installed(cmd):
-            is_installed = self.exec_command(f"type {cmd} >/dev/null 2>&1") == 0
-            if not is_installed:
-                logging.info(f"{cmd} is not installed")
-            return is_installed
-
-        if _is_installed("docker"):
-            docker_magmad_running = self.exec_command_output(
-                "docker ps --filter 'name=magmad' --format '{{.Names}}'",
-            ).strip() == "magmad"
-        else:
-            docker_magmad_running = False
-
-        if _is_installed("systemctl"):
-            systemd_magmad_running = self.exec_command_output(
-                "systemctl is-active magma@magmad",
-            ).strip() == "active"
-        else:
-            systemd_magmad_running = False
+        res_docker = self.exec_command_capture_output(
+            "docker ps --filter 'name=magmad' --format '{{.Names}}'",
+        )
+        res_systemd = self.exec_command_capture_output(
+            "systemctl is-active magma@magmad",
+        )
+        docker_magmad_running = \
+            self._is_installed("docker") & \
```

Comment:
The check if docker/systemd is installed should be done before we check if magmad is running with docker/systemd - and if docker/systemd is not installed then the follow up check should not be executed..

Reply:
Thanks, this was what I actually intended, but I erroneously used `&` instead of `and`. With `and`, the second condition will not be evaluated if the first one is `False`.

```
>>> def func1():
...    print("func1")
...    return False
... 
>>> def func2():
...    print("func2")
...    return False
... 
>>> a = func1() and func2()
func1
>>> a = func1() & func2()
func1
func2
>>> 

```

Reply:
I changed the `&` to `and`, this way the second statement will not be evaluated if the first one is `False`, e.g.:

```
>>> def func1():
...    print("func1")
...    return False
... 
>>> def func2():
...    print("func2")
...    return False
... 
>>> a = func1() and func2()
func1
>>> a = func1() & func2()
func1
func2
>>> 

```

Reply:
@MoritzThomasHuebner I guess the statement of @nstng here means that you need to move `self._is_installed("docker")` and `self._is_installed("systemctl")` checks before line number 934 (before actually triggering shell commands in line numbers 935 and 938 using `exec_command_capture_output`). Or else if you want to move complete check in one like like you did now, you have to trigger shell commands together like below (but this makes command more complex and hence the old code looks simpler here):
```
docker_magmad_running = \
    self._is_installed("docker") and \
    (self.exec_command_capture_output( \
        "docker ps --filter 'name=magmad' --format '{{.Names}}'", \
    ).stdout.decode("utf-8").strip('\n') == 'magmad')
```

Reply:
Sorry, my mistake, it should be fixed now. However, I do have some issues with my VMs right now, so I can't properly test it at the moment.

Reply:
Structure now lgtm. Tested for `systemctl` (with running `magmad` and not running `magmadx`), output and behavior looks good!